### PR TITLE
feat: crypto currency network added

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added a dropdown for crypto_currency networks.

Screenshot 2024-05-28 at 2 40 07 PM

## How did you test it?

Via checking via Crypto Currency Dropdown.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by vsrivatsa-juspay — APPROVED** [review]

File: src/Payments/PaymentMethodsRecord.res [hunk 1]
```diff
@@ -172,7 +173,7 @@ let paymentMethodsFields = [
   },
   {
     paymentMethodName: "crypto_currency",
-    fields: [InfoElement],
+    fields: [CurrencyNetworks, InfoElement],
```
Comment (ArushKapoorJuspay) [inline src/Payments/PaymentMethodsRecord.res:176]: Why are we hardcoding CurrencyNetworks? Shouldn't this data come from backend

File: src/Utilities/PaymentBody.res [hunk 2]
```diff
@@ -547,14 +547,20 @@ let applePayThirdPartySdkBody = (~connectors) => [
...
   ("payment_experience", "redirect_to_url"->JSON.Encode.string),
   (
     "payment_method_data",
     [
-      ("crypto", [("pay_currency", currency->JSON.Encode.string)]->Utils.getJsonFromArrayOfJson),
+      (
+        "crypto",
+        [
+          ("pay_currency", currency->JSON.Encode.string),
+          ("network", cryptoCurrencyNetworks->JSON.Encode.string),
```
Comment (ArushKapoorJuspay) [inline src/Utilities/PaymentBody.res:561]: This should be handled in the dynamic fields

File: src/Components/DynamicFields.res [hunk 3]
```diff
@@ -470,6 +470,7 @@ let make = (
                 optionalRequiredFields={Some(requiredFields)}
               />
             </>
+          | CurrencyNetworks => <CryptoCurrencyNetworks />
```
Comment (ArushKapoorJuspay) [inline src/Components/DynamicFields.res:473]: Could you please rename the enum as CryptoCurrencyNetworks or the file name as CurrencyNetworks. Let's keep this enum either as generic or specific to improve code readability

File: src/Components/CryptoCurrencyNetworks.res [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  )
+
+  let dropdownOptions =
+    Utils.currencyNetworksDict
+    ->Dict.get(currencyVal)
+    ->Option.getOr([])
+    ->Array.map(item => {
+      label: item
+      ->String.toUpperCase
+      ->String.split("_")
```
Comment (ArushKapoorJuspay) [inline src/Components/CryptoCurrencyNetworks.res:17]: Please move this function to Utils next to the other case transformation functions

File: src/Components/DynamicFields.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/Components/DynamicFields.res:473]: Updated

File: src/Components/CryptoCurrencyNetworks.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/Components/CryptoCurrencyNetworks.res]: Updated

File: src/Payments/PaymentMethodsRecord.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/Payments/PaymentMethodsRecord.res]: Updated

File: src/Utilities/PaymentBody.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/Utilities/PaymentBody.res]: Updated

File: src/Utilities/Utils.res [hunk 5]
```diff
@@ -1280,3 +1280,28 @@ let getStateNameFromStateCodeAndCountry = (list: JSON.t, stateCode: string, coun
...
+    ("SHIB", ["ethereum", "bnb_smart_chain"]),
+    ("TRX", ["tron", "bnb_smart_chain"]),
+    ("DOGE", ["dogecoin", "bnb_smart_chain"]),
+    ("BNB", ["bnb_smart_chain"]),
+    ("USDT", ["ethereum", "tron", "bnb_smart_chain"]),
+    ("USDC", ["ethereum", "tron", "bnb_smart_chain"]),
+    ("DAI", ["ethereum", "bnb_smart_chain"]),
+  ]->Dict.fromArray
+
+let splitAndJoinTheString = (~str, ~delimiter) =>
```
Comment (ArushKapoorJuspay) [inline src/Utilities/Utils.res:1303]: Please rename this to toSpacedUpperCase

Comment (PritishBudhiraja) [reply src/Utilities/Utils.res]: Updated

**Review by prafulkoppalkar — APPROVED** [review]

**Review by ArushKapoorJuspay — APPROVED** [review]

**Pull request merged**